addons/mail/static/src/messaging/entity/composer/composer.js

```diff
@@ -188,6 +190,39 @@ function ComposerFactory({ Entity }) {
             return htmlString;
         }
 
+        /**
+         * @private
+         */
+        _generateMentionsLinks(body) {
+            if (this.mentionedPartners.length === 0) {
+                return body;
+            }
+            const inputMentions = body.match(new RegExp("@"+'[^ ]+(?= |&nbsp;|$)', 'g'));
+            const substrings = [];
+            let startIndex = 0;
+            for (const i in inputMentions) {
+                const match = inputMentions[i];
+                const matchName = match.substring(1).replace(new RegExp('\u00a0', 'g'), ' ');
+                const endIndex = body.indexOf(match, startIndex) + match.length;
+                const partner = this.mentionedPartners.find(partner => partner.name === matchName);
+                let mentionLink = "@" + matchName;
+                if (partner) {
+                    const baseHREF = this.env.session.url('/web');
+                    const href = `href='${baseHREF}#model=res.partner&id=${partner.id}' `;
+                    const att_class = `class='o_mail_redirect' `;
+                    const data_oe_id = `data-oe-id='${partner.id}' `;
+                    const data_oe_model = `data-oe-model='res.partner' `;
```

Comment:
camelcase, and end-spaces should not be included in pieces of mention link, but instead be visible in computed string `mentionLink`, as separators when merging the pieces together.